components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts

```diff
@@ -0,0 +1,188 @@
+  templateUrl: './node-credential-details.component.html',
+  styleUrls: ['./node-credential-details.component.scss']
+})
+
+export class NodeCredentialDetailsScreenComponent implements OnInit, OnDestroy {
+  private isDestroyed = new Subject<boolean>();
+  public nodeCredential: NodeCredential;
+  public tabValue: NodeCredentialTabName = 'details';
+  public types: string[] = ['Password', 'RSA'];
+  public passwordSelected = 'Password';
+  public updateForm: FormGroup;
+  public resetForm: FormGroup;
+  public sshForms: FormGroup;
+  public winrmForms: FormGroup;
+  public sudoForms: FormGroup;
+  public id: string;
+  public url: string;
+  public saveSuccessful = false;
+  public saveInProgress = false;
+  public resetSuccessful = false;
+  public resetInProgress = false;
+  public isLoading = true;
+  public nodeDetailsLoading = true;
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    public saveCred: SaveNodeCredential,
+    private fb: FormBuilder,
+    private router: Router,
+    private layoutFacade: LayoutFacadeService
+  ) {
+    this.sshForms = this.fb.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required],
+      key: ['', Validators.required]
+    });
+    this.winrmForms = this.fb.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required]
+    });
+    this.sudoForms = this.fb.group({
+      password: ['', Validators.required],
+      options: ['', Validators.required]
+    });
+
+    this.updateForm = this.fb.group({
+      id: [''],
+      name: ['', Validators.required],
+      type: ['', Validators.required],
+      ssh: this.sshForms,
+      winrm: this.winrmForms,
+      sudo: this.sudoForms
+    });
+
+    this.resetForm = this.fb.group({
+      id: [''],
+      name: ['', Validators.required],
+      type: ['', Validators.required],
+      ssh: this.sshForms,
+      winrm: this.winrmForms,
+      sudo: this.sudoForms
+    });
+    this.getNodeCred();
+  }
+
+  ngOnInit() {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+    this.store.select(routeURL).pipe(takeUntil(this.isDestroyed))
+      .subscribe((url: string) => {
+        this.url = url;
+        const [, fragment] = url.split('#');
+        this.tabValue = (fragment === 'reset') ? 'reset' : 'details';
+      });
+
+    this.store.select(updateStatus).pipe(
+      takeUntil(this.isDestroyed),
+      filter(state => this.saveInProgress || this.resetInProgress && !pending(state)))
+      .subscribe((state) => {
+        if (this.tabValue === 'details') {
+          this.saveInProgress = false;
+          this.saveSuccessful = (state === EntityStatus.loadingSuccess);
+          this.updateForm.markAsPristine();
+        } else {
+          this.resetInProgress = false;
+          this.resetSuccessful = (state === EntityStatus.loadingSuccess);
+          this.resetForm.markAsPristine();
+        }
+        this.getNodeCred();
```

Comment:
Delete this--method is already called on L84 and should only be called once.

Reply:
@msorens Actually we don't have any response after calling the patch Api so for updating the node credential we need to call it as the same happening in creating success.

Reply:
No, this violates subscription lifecycle management. The reason I asked you even to rename it to `setupNodeSubscriptions` is that it is **not** performing any actions--it is simply creating subscriptions and we do not want to setup the same subscription call twice--that simply means the subscription handler will run twice **every** time the data is emitted.